src/java/neroxis/map/SCMap.java

```diff
@@ -376,6 +381,138 @@ public void addCubeMap(CubeMap cubeMap) {
+        repositionEachItemInAnArrayList(heightmapBase, getLargeExpansionAIMarkers(), contentScaler, shiftXAndZ);
+        repositionEachItemInAnArrayList(heightmapBase, getNavalAreaAIMarkers(), contentScaler, shiftXAndZ);
+        repositionEachItemInAnArrayList(heightmapBase, getNavyAIMarkers(), contentScaler, shiftXAndZ);
+        repositionEachItemInAnArrayList(heightmapBase, getLandAIMarkers(), contentScaler, shiftXAndZ);
+        repositionEachItemInAnArrayList(heightmapBase, getNavyRallyMarkers(), contentScaler, shiftXAndZ);
+        repositionEachItemInAnArrayList(heightmapBase, getRallyMarkers(), contentScaler, shiftXAndZ);
+        repositionEachItemInAnArrayList(heightmapBase, getBlankMarkers(), contentScaler, shiftXAndZ);
+        repositionEachItemInAnArrayList(heightmapBase, getHydros(), contentScaler, shiftXAndZ);
+        repositionEachItemInAnArrayList(heightmapBase, getMexes(), contentScaler, shiftXAndZ);
+        repositionEachItemInAnArrayList(heightmapBase, getProps(), contentScaler, shiftXAndZ);
+
+        for (int i = 0; i < getDecalCount(); i++) {
+            float x = (float) (StrictMath.round((contentScaler * getDecal(i).getPosition().x + shiftX) + 0.5) - 0.5);
+            float z = (float) (StrictMath.round((contentScaler * getDecal(i).getPosition().z + shiftZ) + 0.5) - 0.5);
+            float y;
+            if(heightmapBase.inBounds((int) x,  (int) z)) {
+                y = heightmapBase.getValueAt((int) x, (int) z);
+            } else {
+                y = 0; }
+            getDecal(i).setPosition(new Vector3f(x, y, z));
+            Vector3f scale = getDecal(i).getScale();
+            getDecal(i).setScale(new Vector3f(scale.x * contentScaler, scale.y, scale.z * contentScaler));
+            getDecal(i).setCutOffLOD(getDecal(i).getCutOffLOD() * contentScaler);
+        }
+
+        for (int i = 0; i < getArmyCount(); i++) {
+            Army army = getArmy(i);
+            for (int a = 0; a < army.getGroupCount(); a++) {
+                Group group = army.getGroup(a);
+                repositionEachItemInAnArrayList(heightmapBase, group.getUnits(), contentScaler, shiftXAndZ);
+            }
+        }
+    }
+
+    public void repositionEachItemInAnArrayList(FloatMask heightmapBase, ArrayList<? extends PositionedObject> arrayListOfPositionedObjects, float distanceScaler, Vector2f shiftXAndZ) {
+        for (PositionedObject positionedObject : arrayListOfPositionedObjects) {
+            float x = (float) (StrictMath.round((distanceScaler * positionedObject.getPosition().x + shiftXAndZ.x) + 0.5) - 0.5);
+            float z = (float) (StrictMath.round((distanceScaler * positionedObject.getPosition().z + shiftXAndZ.y) + 0.5) - 0.5);
+            float y;
+            if (heightmapBase.inBounds((int) x, (int) z)) {
+                y = heightmapBase.getValueAt((int) x, (int) z);
+            } else {
+                y = 0;
+            }
+            positionedObject.setPosition(new Vector3f(x, y, z));
```

Comment:
It should be possible to remove the dependency of heightmapBase in this function. 
The Placement class has a function placeOnHeightmap which will do lines 507-510. Additionally Vector2f has an add and subtract method. Would just need to add a multiply method and then you wouldn't need to break out the x and z components. 
Vector2f and Vector3f also have round to nearestHalfPoint method